### ISSUE TITLE
Question on Efficientnet notebook example

### ISSUE THREAD

**Issue by Soo95** [pytorch/ignite#594]
Hi, I started to learn ignite a few days ago.
I have a question on efficientnet notebook example.

why is 'eval_train_loader' created for 'train_evaluator'?

is there a problem if I put 'train_loader' as train_evaluator instead of 'eval_train_loader'?
I'm curious why two train_loader are needed.

**Comment by vfdev-5** [comment]
Hi @Soo95, 

> why is 'eval_train_loader' created for 'train_evaluator'?

Idea is to compute metrics on a train subset and on the test dataset.
Train subset (by `eval_train_loader`) is smaller than the train dataset and not shuffled in the loader. And if we would like to make it more clean, train subset dataset transformations should be as for the test dataset.

> is there a problem if I put 'train_loader' as train_evaluator instead of 'eval_train_loader'?

There is no problem with that. However, metrics computation can take more time and as it has random data augs, metric values wont be exactly the same at multiple runs.

**Comment by Soo95** [comment]
@vfdev-5 
Thanks a lot! I understood why it is need.
